code/game/objects/items/defib.dm

```diff
@@ -451,11 +451,15 @@
 	return TRUE
 
 /obj/item/twohanded/shockpaddles/proc/shock_touching(dmg, mob/H)
-	if(isliving(H.pulledby))		//CLEAR!
-		var/mob/living/M = H.pulledby
-		if(M.electrocute_act(30, src))
-			M.visible_message("<span class='danger'>[M] is electrocuted by [M.p_their()] contact with [H]!</span>")
-			M.emote("scream")
+	if(!ishuman(H.pulledby))		//CLEAR!
+		return
+	var/mob/living/carbon/human/M = H.pulledby
+	if(M.has_trait(TRAIT_SHOCKIMMUNE) || (M.gloves && M.gloves.siemens_coefficient <= 0) || M.dna.species.siemens_coeff <= 0)
```

Comment:
These should be covered by carbon `electrocute_act`

Reply:
This one is an exception to the electrocute_act rule. It needs the extra checks to stop pulling before getting shocked, to prevent shocking the defibee.

Reply:
You right